common/upload.js

```diff
@@ -84,13 +85,16 @@
                     scope.setFilename = function () {
                         var value = scope.value;
                         if (value) {
-                            if (value.filename) {
-                                return value.filename;
-                            } else {
-                                return value.url;
-                            }
+                            // value.filename will always be either the stored filename or the "caption" (text stripped of the hatrac path)
+                            return value.filename;
                         }
                     }
+
+                    scope.fileTooltip = function () {
+                        var value = scope.value;
+                        // value.filename will always be either the stored filename or the "caption" (text stripped of the hatrac path)
+                        return (value.filesize ? "- " + value.filename + "<br>- " + value.filesize + " bytes" : value.filename);
```

Comment:
You should call `humanFileSize` function that we have in utils.js. Instead of showing the raw byte count.

Also I'm not sure whether we want to handle this or even how we should handle this. But the `value.filename` won't be available in case of default values in create mode. We're calling into `formatFile` function which only is attaching `url` to the value. If we want to attach the `filename` by getting the caption from ermrestjs then we have to create a key-value pair of all the default values of the table. Or maybe just passing the url value itself is enough and ermrestjs will return the last part of url for you.

Reply:
> Or maybe just passing the url value itself is enough and ermrestjs will return the last part of url for you.

That's what I was thinking. The generate caption function could be public on the `ermrestJS` object in `chaise` so it could be used in a similar way to `renderTemplates`. i.e. `ErmrestJS.getFileCaption(defaultUrl)`.

Reply:
We don't need any extra apis. the `getMetadata` can be used here too. You just have to pass an object with just the url column default value. ermrestjs should handle it. 

You can ignore the "we have to create a key-value pair of all the default values of the table". I was thinking about a case that shouldn't even happen. The case that I was thinking was when you have defined meta columns in your annotation. What I assumed is that in this case for caption we should use the default value of `filename` column. But I don't think that's the behavior that we want. If we want to do this, then we should also send the default value of filename to database (if users don't change the value of that file).

Reply:
Oh right. I forgot `getMetadata` was written to accept an argument.

And yes, if we have a case where the asset column and filename column both have default values, we should honor them as is. The data modeler and user would want to see what the `asset` path is alongside the filename if both had defaults and were visable but diabled during create.